datalad/runner/runner.py

```diff
@@ -70,7 +72,7 @@ def run(self,
             env=None,
             timeout=None,
             exception_on_error=True,
-            **kwargs):
+            **kwargs) -> dict | Generator:
```

Comment:
@christian-monch it seems we have some possible non-uniformity.  Docstring here says `Union[Any, Generator]` but then nonasyncrunner..run has `def run(self) -> dict | Generator:` .  I wondered what it truly be -- `Any` sounds a bit too unspecific.

@jwodder in such cases where we have a composite type present/used in many spots, should we define some type to be reused?

Reply:
If it's just two short things like `dict | Generator`, I don't really see the need.

Reply:
I agree with @jwodder. Firstly, as he points out, it is just a short annotation, secondly, a telling name might not be shorter, and thirdly, we might clutter the name space if we define a lot of custom types

Reply:
@yarikoptic: yes, the docstring is wrong. We should put `dict | Generator` as return type